### ISSUE TITLE
Add text on nested or pattern 

### ISSUE THREAD

**Issue by cuihtlauac** [ocaml/ocaml.org#1977]
This is valid OCaml:
```ocaml
# function Ok Either.(Left x | Right x) -> Some x | Error _ -> None;;
- : (('a, 'a) Either.t, 'b) result -> 'a option = <fun>

# let any (Either.Left x | Either.Right x) = x;;
val any : ('a, 'a) Either.t -> 'a = <fun>
```

It should be mentioned in the Basic Data Types and Pattern-Matching tutorial. Thanks @chshersh for point this

**Comment by The-Amoghavarsha** [comment]
Interesting. Does it suffice, if we add this in the https://ocaml.org/docs/basic-data-types#a-complete-example-mathematical-expressions sub-section?

**Comment by cuihtlauac** [comment]
It would be nice to have text about this there. But a section on nested matching is needed. We'd also discuss matching on several types; it will include the above but won't be limited to it.

**Comment by cuihtlauac** [comment]
Maybe we should do a whole document on advanced pattern matching
- This
- Combined patterns, expressions with several data types
- `as` patterns
- `when` patterns
- stacked/fallback patterns
- exceptions patterns

**Comment by sabine** [comment]
Yes, I think this goes beyond the scope of Basic Data Types and Pattern-Matching and having a separate document would be great!
